doc/tutorials/contributing-clion.md

```diff
@@ -0,0 +1,183 @@
+  almost all pages of the [homepage](https://www.libelektra.org) of this
+  project. One important note is, that all markdown-pages can also be used
+  for testing using
+  [Markdown to Shell Recorder](https://github.com/ElektraInitiative/libelektra/tree/master/tests/shell/shell_recorder/tutorial_wrapper)
+  (you can find an example on how to do this [here](/doc/help/kdb-get.md)).
+- src<br/>
+  Almost all functionality-code resides here.
+  - bindings<br/>
+    Here is all the code of available [Bindings](/src/bindings/README.md)
+    of libelektra. - plugins<br/>
+    You can find all developed [Plugins](/src/plugins/README.md) here. If
+    you want to fix a bug for an existing plugin or add another one, this is
+    the directory you have to work in. For further information on how to
+    develop your own plugin, please visit [this](/doc/tutorials/plugins.md)
+    tutorial. - tools<br/>
+    In this folder the source of all tools for interacting with Electra's
+    global key database, e.g. _kdb_, can be found.
+- tests<br/>
+  Here you can find all sorts of tests (excluding tests created with the
+  _Markdown to Shell Recorder_-tool).
+
+## Testing
+
+The most thorough way to test your changes is to run all tests. Therefore
+navigate to your run-configurations (Run -> Edit Configurations...) and look
+for the entry "run*all". There, "run_all" should be selected as \_Executable*,
+"kdb" as _Target_. Now you can execute this run-configuration which will run
+all enabled tests!
+
+If you want to test various _kdb_-methods separately, you can create your
+own run-configurations. Add a new one by clicking on the "+"-sign on the
+top left of the "Edit Configurations..."-window and name it. Here both
+_Executable_ and _Target_ should have "kdb" selected. If you for example
+want to test `kdb info dump`, write "info dump" next to _Program arguments_.
+That's it, now you can just test this part of _kdb_.
+
+For further information please read [this](/doc/TESTING.md).
+
+## Commiting your changes
+
+Once you are satisfied with your changes, you have to commit them to your
+forked repository. We'll use the terminal for working with git. By
+convention, such commits shouldn't be too large, otherwise it will become
+difficult to revert some small changes if they are not working as intended.
+To be able to commit code to your repository, you have to configure your
+local git-installation to use your GitHub-credentials. You can find
+information about how to do this
+[here](https://help.github.com/en/articles/set-up-git).
+
+After you've set up your git-configuration, you can continue with uploading
+your changes. By default, you are in the master-branch of your repository.
+First, you should never directly work on the master branch, since only
+working code is expected to be there. This means, we now create a branch in
+our repository where our code changes will be published into. To do this,
```

Comment:
In a Clion tutorial you should really describe how to do this using the built in Git (and GitHub) support.

Also add:
Ensure that "Reformat code" and "Rearrange code" are disabled in the commit dialog. Otherwise Clions formatter might produce files that don't adhere to our formatting guidelines.
If you installed the `pre-commit-check-formatting` pre-commit-hook from the `scripts` directory. Ensure that "Run Git hooks" is enabled in the commit dialog.